app/components/Authentication/AwsEc2/AwsEc2.jsx

```diff
@@ -1,27 +1,450 @@
+                    configObj: update(this.state.configObj,
+                        {
+                            access_key: { $set: (config.access_key ? config.access_key : null) },
+                            endpoint: { $set: config.endpoint },
+                            secret_key: { $set: (config.secret_key ? config.secret_key : null) }
+                        }),
+                    newConfigObj: update(this.state.configObj,
+                        {
+                            access_key: { $set: (config.access_key ? config.access_key : null) },
+                            endpoint: { $set: config.endpoint },
+                            secret_key: { $set: (config.secret_key ? config.secret_key : null) }
+                        })
+                });
+            })
+            .catch(snackBarMessage);
+    }
+
+    createUpdateConfig() {
+        callVaultApi('post', `${this.state.baseVaultPath}/config/client`, null, this.state.newConfigObj)
+            .then(() => {
+                snackBarMessage(`Backend ${this.state.baseVaultPath}/config has been updated`);
+            })
+            .catch(snackBarMessage);
     }
+
+    createUpdateRole() {
+        let updateObj = _.clone(this.state.newRoleConfig);
+        updateObj.policies = updateObj.policies.join(',');
+        callVaultApi('post', `${this.state.baseVaultPath}/role/${this.state.newRoleConfig.role}`, null, updateObj)
+            .then(() => {
+                snackBarMessage(`Role ${this.state.newRoleConfig.role} has been updated`);
+                this.listEc2Roles();
+                this.setState({ openNewRoleDialog: false, openEditRoleDialog: false, newRoleConfig: _.clone(this.roleConfigSchema) });
+            })
+            .catch(snackBarMessage);
+    }
+
+    displayRole() {
+        tokenHasCapabilities(['read'], `${this.state.baseVaultPath}/role/${this.state.newRoleConfig.role}`)
+            .then(() => {
+                callVaultApi('get', `${this.state.baseVaultPath}/role/${this.state.newRoleConfig.role}`, null, null, null)
+                    .then((resp) => {
+                        resp.data.data.role = this.state.newRoleConfig.role;
+                        this.setState({ newRoleConfig: resp.data.data, openEditRoleDialog: true });
+                    })
+                    .catch(snackBarMessage)
+            })
+            .catch(() => {
+                this.setState({ selectedUserObject: {} })
+                snackBarMessage(new Error(`No permissions to display properties for role ${this.state.newRoleConfig.role}`));
+            })
+    }
+
+    componentDidMount() {
+        this.listEc2Roles();
+        this.getEc2AuthConfig();
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (this.state.newRoleConfig.role != prevState.newRoleConfig.role) {
+            this.listEc2Roles();
+            if (this.state.newRoleConfig.role) {
+                this.displayRole();
+            }
+        }
+    }
+
```

Comment:
This is missing code to handle switching to another aws-ec2 backend mounted on a different path.

Refer to `app/components/Authentication/Radius/Radius.jsx:componentWillReceiveProps()`